apl-core/src/main/java/com/apollocurrency/aplwallet/apl/eth/service/EthereumWalletService.java

```diff
@@ -47,31 +57,94 @@
 
     private String paxContractAddress = propertiesHolder.getStringProperty("apl.eth.pax.contract.address");
 
-    public BigInteger getPaxBalanceWei(String accountAddress){
-        return getTokenBalance(paxContractAddress, accountAddress);
+    /**
+     * Get balances for Eth/tokens.
+     * @param address Eth address
+     * @return account balance in Wei
+     */
+    public EthWalletBalanceInfo balanceInfo(String address){
+        EthWalletBalanceInfo ethWalletBalanceInfo = new EthWalletBalanceInfo(address);
+
+        ethWalletBalanceInfo.put(DexCurrencies.ETH.getCurrencyCode(), getEthBalanceWei(address));
+        ethWalletBalanceInfo.put(DexCurrencies.PAX.getCurrencyCode(), getPaxBalanceWei(address));
+
+        return ethWalletBalanceInfo;
+    }
+
+    /**
+     * Get Eth PAX token balance.
+     * @param address Eth address
+     * @return account balance in Wei
+     */
+    public BigInteger getPaxBalanceWei(String address){
+        return getTokenBalance(paxContractAddress, address);
+    }
+
+    /**
+     * Get Eth balance.
+     * @param address Eth address
+     * @return account balance in Wei
+     */
+    public BigInteger getEthBalanceWei(String address) {
```

Comment:
That service class can be potentially called not only from Controller, but it can be potentially called from another Service. It would be better to put simplest checks for incoming arguments on most public methods.

I know most of checks are in controller.
What is result of such call
`web3j.ethGetBalance(address, DefaultBlockParameterName.LATEST)` if address is NULL ?

Objects.requireNonNull(...) are very cheap to put into code and they can prevent troubles if it will be called (someday) from another service class. IMHO.